### ISSUE TITLE
FailedPreconditionError (see above for traceback): Attempting to use uninitialized value 

### ISSUE THREAD

**Issue by diemtran1005** [tensorflow/tensorflow#28605]
FailedPreconditionError (see above for traceback): Attempting to use uninitialized value con1/bias
         [[node con1/bias/read (defined at c:/users/tran thi diem/documents/diem/research/reinforcementlearning/code/codeforlearningreinforcement/cnnandrl/object-recognition-cifar-10-master/code/codechuan/10_5_2019/2_5_2019_cifar_modify_bangchay_testingwithrl.py:60) ]]

**System information**

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Window10

- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): 1.13.1
- Python version: 3.6
**Describe the current behavior**
I have this bug but I don't know how to fix this
**Describe the expected behavior**

**Code to reproduce the issue**
import tensorflow as tf
from keras.datasets import cifar10
from keras.utils import np_utils
from keras.preprocessing.image import ImageDataGenerator
#from tesnsorlow import keras

#import random
import numpy as np
from scipy.stats import spearmanr

from sklearn.utils import shuffle

def lr_schedule(epoch):
    lrate = 0.001
    if epoch > 75:
        lrate = 0.0005
    elif epoch > 100:
        lrate = 0.0003        
    return lrate
class DQNetwork:
    
    def __init__(self):
#        self.state_size = state_size
 #       self.action_size = action_size
        self.action = tf.placeholder(tf.float32)
        if (self.action == 0):
            test = 5
        else:
            test = 7
            
        sess = tf.Session()
        sess.run(tf.global_variables_initializer())
        self.learning_rate = tf.placeholder(tf.float32)

        self.inputs_ = tf.placeholder(tf.float32, [None, *state_size])       

        self.label = tf.placeholder(tf.float32, [None,10])
        

        self.filter_size_layer1 = tf.placeholder(tf.float32) 
        self.filter_size_layer2 = tf.placeholder(tf.float32) 
        self.filter_size_layer3 = tf.placeholder(tf.float32) 
        self.filter_size_layer4 = tf.placeholder(tf.float32) 
        self.filter_size_layer5 = tf.placeholder(tf.float32) 
        self.filter_size_layer6 = tf.placeholder(tf.float32)
        ###############
#layer 1
        # Input is 84x84x4 
        self.conv1 = tf.layers.conv2d(inputs=self.inputs_,
                                      filters=1,
                                      kernel_size=[test,test],
                                     
                                      #kernel_size=[3, 3],
                                      #kernel_size=[self.filter_size_layer1,self.filter_size_layer1],
                                      strides=[1,1],
                                      padding="same",  
                                      kernel_regularizer=tf.contrib.layers.l2_regularizer(1e-4),
                                      name = "con1",
                                      reuse=tf.AUTO_REUSE
                                      #kernel_regularizer=regularizers.l2(1e-4)
                                      )
       
       
        self.weight_conv1 = tf.all_variables()
        
        
        self.conv1_activation = tf.nn.elu(self.conv1)

        self.conv1_batchnorm = tf.layers.batch_normalization(self.conv1_activation,
                                                             training=True,
                                                             epsilon=1e-5
                                                             )
        
       
      
#layer2

        self.conv2 = tf.layers.conv2d(inputs=self.conv1_batchnorm,
                                      filters=32,
                                      kernel_size=[3, 3],
                                      strides=[1,1],
                                      padding="same",
                                      kernel_regularizer=tf.contrib.layers.l2_regularizer(1e-4),
                                      name = "con2",
                                      reuse=tf.AUTO_REUSE
                                      #kernel_regularizer=regularizers.l2(1e-4)
                                      )
        self.conv2_activation = tf.nn.elu(self.conv2)
        self.conv2_batchnorm = tf.layers.batch_normalization(self.conv2_activation,
                                                             training=True,
                                                             epsilon=1e-5
                                                             )
        self.conv2_Maxpool = tf.layers.max_pooling2d(self.conv2_batchnorm, 1, 2)
        #self.conv2_Maxpool = tf.layers.MaxPooling2D(self.conv2_batchnorm, pool_size =(2,2)) 
        self.conv2_Dropout = tf.layers.dropout (self.conv2_Maxpool,rate = 0.2)
        

#layer3
        self.conv3 = tf.layers.conv2d(inputs=self.conv2_Dropout,
                                      filters=64,
                                      kernel_size=[3,3],
                                      strides=[1,1],
                                      padding="same",
                                      kernel_regularizer=tf.contrib.layers.l2_regularizer(1e-4),
                                      name = "con3",
                                      reuse=tf.AUTO_REUSE
                                      
                                      #kernel_regularizer=regularizers.l2(1e-4)
                                      )
        self.conv3_activation = tf.nn.elu(self.conv3)

        self.conv3_batchnorm = tf.layers.batch_normalization(self.conv3_activation,
                                                             training=True,
                                                             epsilon=1e-5
                                                             )
#layer 4
       

        self.conv4 = tf.layers.conv2d(inputs=self.conv3_batchnorm,
                                      filters=64,
                                      kernel_size=[3, 3],
                                      strides=[1,1],
                                      padding="same",
                                      kernel_regularizer=tf.contrib.layers.l2_regularizer(1e-4),
                                      reuse=tf.AUTO_REUSE,
                                      name = "con4"
                                      
                                      #kernel_regularizer=regularizers.l2(1e-4)
                                      )
        
        self.conv4_activation = tf.nn.elu(self.conv4)
        self.conv4_batchnorm = tf.layers.batch_normalization(self.conv4_activation,
                                                             training=True,
                                                             epsilon=1e-5
                                                             )
        self.conv4_Maxpool = tf.layers.max_pooling2d(self.conv4_batchnorm, 1, 2)

        #self.conv4_Maxpool = tf.layers.MaxPooling2D(self.conv4_batchnorm, pool_size =(2,2)) 
        self.conv4_Dropout = tf.layers.dropout (self.conv4_Maxpool,rate = 0.3)
        
        
#layer5
        self.conv5 = tf.layers.conv2d(inputs=self.conv4_Dropout,
                                      filters=128,
                                      kernel_size=[3,3],
                                      strides=[1,1],
                                      padding="same",
                                      kernel_regularizer=tf.contrib.layers.l2_regularizer(1e-4),
                                      reuse=tf.AUTO_REUSE,
                                      name = "con5"
                                     # kernel_regularizer=regularizers.l2(1e-4)
                                      )
        self.conv5_activation = tf.nn.elu(self.conv5)

        self.conv5_batchnorm = tf.layers.batch_normalization(self.conv5_activation,
                                                             training=True,
                                                             epsilon=1e-5
                                                             )
#layer 6
       

        self.conv6 = tf.layers.conv2d(inputs=self.conv5_batchnorm,
                                      filters=128,
                                      kernel_size=[3, 3],
                                      strides=[1,1],
                                      padding="same",
                                      kernel_regularizer=tf.contrib.layers.l2_regularizer(1e-4),
                                      reuse=tf.AUTO_REUSE,
                                      name = "con6"
                                     # kernel_regularizer=regularizers.l2(1e-4)
                                      )
        self.conv6_activation = tf.nn.elu(self.conv6)
        self.conv6_batchnorm = tf.layers.batch_normalization(self.conv6_activation,
                                                             training=True,
                                                             epsilon=1e-5
                                                             )
        self.conv6_Maxpool = tf.layers.max_pooling2d(self.conv6_batchnorm, 1, 2)

       # self.conv6_Maxpool = tf.layers.MaxPooling2D(self.conv6_batchnorm, pool_size =(2,2)) 
        self.conv6_Dropout = tf.layers.dropout (self.conv6_Maxpool,rate = 0.4)   
      
          
#fully connected        
        
           

        self.flatten = tf.layers.flatten(self.conv6_Dropout)
        ## --> [43,264]

        self.output = tf.layers.dense(inputs=self.flatten,
                                      kernel_regularizer=tf.contrib.layers.l2_regularizer(1e-4),
                                      #kernel_regularizer=regularizers.l2(1e-4),
                                      units=10,
                                      activation='softmax')
        
        self.targets1 = tf.squeeze(tf.cast(self.label, tf.int32))   # Get predicted values by finding which logit is the greatest
        self.targets = tf.cast(tf.argmax(self.targets1, 1), tf.int32)        
        self.batch_predictions = tf.cast(tf.argmax(self.output, 1), tf.int32)
        self.predicted_correctly = tf.equal(self.batch_predictions, self.targets)
    # Average the 1's and 0's (True's and False's) across the batch size
        self.accuracy = tf.reduce_mean(tf.cast(self.predicted_correctly, tf.float32))
        
        
        self.cross_entropy =  tf.nn.softmax_cross_entropy_with_logits_v2(logits= self.output, labels=self.label)
        self.gradient = tf.gradients (self.cross_entropy,self.weight_conv1)  
        self.loss = tf.reduce_mean( tf.nn.softmax_cross_entropy_with_logits_v2(logits= self.output, labels=self.label) )
          
        #cross_entropy = tf.nn.sparse_softmax_cross_entropy_with_logits_v2(logits = self.output,label = self.label)
       # self.loss = tf.reduce_mean(cross_entropy, name='cross_entropy')
        self.optimizer = tf.train.RMSPropOptimizer(self.learning_rate,decay=1e-6).minimize(self.loss)
        
        sess = tf.Session()
        sess.run(tf.global_variables_initializer())
        
        
       
        #self.optimizer = tf.train.AdamOptimizer(self.learning_rate,decay=1e-6).minimize(self.loss)

(X_train, y_train), (X_test, y_test) = cifar10.load_data()
#X_train = X_train.astype('float32')
#X_test = X_test.astype('float32')
#X_train = normalize(X_train)
#X_test = normalize(X_test)
#y_train = np_utils.to_categorical(y_train)
#y_test = np_utils.to_categorical(y_test)
#num_classes = y_test.shape[1]
X_train = X_train.astype('float32')
X_test = X_test.astype('float32')

#z-score
mean = np.mean(X_train,axis=(0,1,2,3))
std = np.std(X_train,axis=(0,1,2,3))
X_train = (X_train-mean)/(std+1e-7)
X_test = (X_test-mean)/(std+1e-7)

num_classes = 10
y_train = np_utils.to_categorical(y_train,num_classes) # convert the numbers to onehot vector
y_test = np_utils.to_categorical(y_test,num_classes)
#DQNetwork

action_size = 3
#learning_rate = 0.00001

state_size = [32,32,3]

# Exploration parameters for epsilon greedy strategy
explore_start = 1.0            # exploration probability at start
explore_stop = 0.01            # minimum exploration probability 
           # exponential decay rate for exploration prob

# Q learning hyperparameters
gamma = 0.95  
alpha = 0.5

 
sess = tf.Session()
sess.run(tf.global_variables_initializer())

Qtable_before =np.zeros((12500,3))
np.random.seed(100)
exp_exp_tradeoff = 0.25

generations = 20
batch = 64

def train_process (X_train,y_train,batch,before_loss_train,learnig_rate):
  #  gradient_descent_test = []
    number_image = 0   
    
    while (number_image <len(X_train[0:64])):
        before_state = X_train[number_image: number_image +batch]
        before_Y_label = y_train [number_image: number_image +batch]
   #while i<20:
               
        _,current_loss,current_accuracy = sess.run([dqn.optimizer,dqn.loss,dqn.accuracy],
                                                   feed_dict={dqn.inputs_:before_state,dqn.label:before_Y_label,dqn.learning_rate:learning_rate})
     
 
        number_image = number_image + batch   
            
               
        
       
    return current_loss,current_accuracy
        
        #print("step {}: eps {} : loss {} ".format (i,eps, current_loss))
train_loss = []
train_accuracy = []
test_loss = []
test_accuracy = []
before_loss_train = 100
eps =0

datagen = ImageDataGenerator(
    rotation_range=15,
    width_shift_range=0.1,
    height_shift_range=0.1,
    horizontal_flip=True,
    )
datagen.fit(X_train)

number_image = 0   
learning_rate = 0.1

gradient_total = []
shape = []

i =0
action = 1
dqn = DQNetwork()
    
while (number_image <len(X_train[0:640])):
    before_state = X_train[number_image: number_image +batch]
    before_Y_label = y_train [number_image: number_image +batch]
   # test = sess.run(dqn.test,feed_dict = {dqn.action:action})
    conv1 = sess.run([dqn.conv1],feed_dict = {dqn.inputs_: before_state,dqn.action:action})
   #while i<20:
    weight = sess.run(dqn.weight_conv1,feed_dict = {dqn.inputs_: before_state,dqn.action:action})       
    gradient = sess.run (dqn.gradient,feed_dict={dqn.inputs_:before_state,dqn.label:before_Y_label,dqn.learning_rate:learning_rate,dqn.action:action})
    _,current_loss,current_accuracy = sess.run([dqn.optimizer,dqn.loss,dqn.accuracy],
                                               feed_dict={dqn.inputs_:before_state,dqn.label:before_Y_label,dqn.learning_rate:learning_rate,dqn.action:action})
    number_image = number_image + batch
    gradient = np.mean (gradient [0])
    gradient_total.append(gradient)
    shape.append (np.array (weight[0]).shape)
    i = i+1
#    print ("\n weigth",weight [0])
#    print ("\n bias",weight [1])
#    print ("\n gradient weight", np.mean (gradient[0]))
#    print ("\n gradient bias", gradient[1])
    print("step {}: loss {} : accuracy {} ".format (i, current_loss,current_accuracy))
    
    tf.reset_default_graph()
print (gradient_total)   
**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

**Comment by achandraa** [comment]
Will it be possible to provide a minimal code snippet that can replicate the issue since it will be easier to debug and decide on the same. Thanks!

**Comment by achandraa** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
